presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2802,6 +2802,306 @@ public void testCollectColumnStatisticsOnInsert()
+        assertQueryFails(format("ANALYZE %s WITH (partitions = 1)", tableName), ".*Cannot convert.*to array\\(array\\(varchar\\)\\).*");
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = NULL)", tableName), ".*Invalid null value for analyze property.*");
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[NULL])", tableName), ".*Invalid null value in analyze partitions property.*");
+
+        // Test non-existed partition
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p4']])", tableName), ".*Partition no longer exists: p_varchar=p4.*");
+
+        // Test partition schema mismatch
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p4', 'error']])", tableName), ".*Partition value count does not match the partition column count.*");
+
+        // Drop the partitioned test table
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testInvalidAnalyzeUnpartitionedTable()
+    {
+        String tableName = "test_invalid_analyze_unpartitioned_table";
+
+        // Test table does not exist
+        assertQueryFails("ANALYZE " + tableName, ".*Table.*does not exist.*");
+
+        createUnpartitionedTableForAnalyzeTest(tableName);
+
+        // Test partition properties on unpartitioned table
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[])", tableName), ".*Only partitioned table can be analyzed with a partition list.*");
+        assertQueryFails(format("ANALYZE  %s WITH (partitions = ARRAY[ARRAY['p1']])", tableName), ".*Only partitioned table can be analyzed with a partition list.*");
+
+        // Drop the partitioned test table
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testAnalyzePartitionedTable()
+    {
+        String tableName = "test_analyze_partitioned_table";
+        createPartitionedTableForAnalyzeTest(tableName);
+
+        // No column stats before running analyze
+        assertQuery(format("SHOW STATS FOR (SELECT * FROM %s)", tableName),
+                "SELECT * FROM VALUES " +
+                        "('c_boolean', null, null, null, null, null, null), " +
+                        "('c_bigint', null, null, null, null, null, null), " +
+                        "('c_double', null, null, null, null, null, null), " +
+                        "('c_timestamp', null, null, null, null, null, null), " +
+                        "('c_varchar', null, null, null, null, null, null), " +
+                        "('c_varbinary', null, null, null, null, null, null), " +
+                        "('p_varchar', 24.0E0, 6.0E0, 0.25E0, null, null, null), " +
+                        "(null, null, null, null, 16.0E0, null, null)");
+
+        // No column stats after running an empty analyze
+        assertUpdate(format("ANALYZE %s WITH (partitions = ARRAY[])", tableName));
```

Comment:
Shouldn't this return 0?

Reply:
Yes. I fixed the logic in `AnalyzeFinishOperator` so that it will return 0 now. Also fixed the test.